pyoozie/tags.py

```diff
@@ -0,0 +1,317 @@
+        Xml.__init__(self, 'sub-workflow')
+        self.app_path = app_path
+        self.propagate_configuration = propagate_configuration
+        self.configuration = Configuration(configuration)
+
+    def _xml(self, doc, tag, text):
+        with tag(self.xml_tag):
+            with tag('app-path'):
+                doc.text(self.app_path)
+            if self.propagate_configuration:
+                doc.stag('propagate-configuration')
+            if self.configuration:
+                self.configuration._xml(doc, tag, text)
+
+        return doc
+
+
+class GlobalConfiguration(Xml):
+    """Global configuration values for all actions in a workflow.
+
+    "Oozie allows a global section to reduce the redundant job-tracker and name-node declarations for each action.
+    [...] The global section may contain the job-xml, configuration, job-tracker, or name-node that the user would
+    like to set for every action.  If a user then redefines one of these in a specific action node, Oozie will
+    update [sic] use the specific declaration instead of the global one for that action."
+
+    "The job-xml element, if present, must refer to a Hadoop JobConf job.xml file bundled in the workflow
+    application."
+    """
+
+    def __init__(self, job_tracker='', name_node='', job_xml_files=None, configuration=None):
+        Xml.__init__(self, 'global')
+        self.job_tracker = job_tracker
+        self.name_node = name_node
+        self.job_xml_files = job_xml_files if job_xml_files else list()
+        self.configuration = Configuration(configuration)
+
+    def _xml(self, doc, tag, text):
+        with tag(self.xml_tag):
+            if self.job_tracker:
+                with tag('job-tracker'):
+                    doc.text(self.job_tracker)
+            if self.name_node:
+                with tag('name-node'):
+                    doc.text(self.name_node)
+            if self.job_xml_files:
+                for xml_file in self.job_xml_files:
+                    with tag('job-xml'):
+                        doc.text(xml_file)
+            if self.configuration:
+                self.configuration._xml(doc, tag, text)
+
+        return doc
+
+    def __nonzero__(self):
+        return any(bool(x) for x in [
+            self.job_tracker, self.name_node, self.job_xml_files,
+            self.configuration
+        ])
+
+
+class Email(Xml):
+    """Email action for use within a workflow."""
+
+    def __init__(self, to, subject, body, cc=None, bcc=None, content_type=None, attachments=None):
+        Xml.__init__(self, 'email')
+        self.to = to
+        self.subject = subject
+        self.body = body
+        self.cc = cc
+        self.bcc = bcc
+        self.content_type = content_type
+        self.attachments = attachments
+
+    def _xml(self, doc, tag, text):
+        def format_list(emails):
+            return ','.join(sorted(emails)) if hasattr(emails, '__iter__') and not \
+                isinstance(emails, str) else emails
```

Comment:
Given that you've rolled onto two lines, should this _really_ need to be a single statement?